web/src/main/java/se/inera/intyg/webcert/web/integration/interactions/createdraftcertificate/BaseCreateDraftCertificateValidator.java

```diff
@@ -182,7 +180,8 @@ private void validateSekretess(ResultValidator errors, String intygsTyp, Sekrete
         if (!authoritiesHelper.getIntygstyperAllowedForSekretessmarkering().contains(intygsTyp)) {
             switch (sekretessStatus) {
                 case TRUE:
-                    errors.addError("Intygstypen {0} kan inte utfärdas för patienter med skyddade personuppgifter.", intygsTyp);
+                    errors.addError("Intygstypen {0} kan inte utfärdas för patienter med skyddade personuppgifter.",
+                        getCertificateDisplayName(intygsTyp));
```

Comment:
Instead of hard-coding you could use the IntygModuleRegistry e.g. moduleRegistry.getModuleEntryPoint(certificateType).getModuleName().